src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -118,6 +120,9 @@ public bool HasLazyStaticConstructor(TypeDesc type)
 
         public MethodDebugInformation GetDebugInfo(MethodIL methodIL)
         {
+            if (!GenerateDebugInfo)
```

Comment:
What's the scenario for not generating debug info? On other runtimes, optimized builds are still debuggable to some extent.